### ISSUE TITLE
Support abstract domain sockets using epoll on linux

### ISSUE THREAD

**Issue by yawkat** [netty/netty#4085]
Version: 5.0.0.Alpha2

Context:
I am trying to write a client for dbus on linux and need to support abstract domain sockets to properly connect to the native dbus servers.

It is currently not possible to create an abstract domain socket using the epoll selector. While creating normal domain sockets works fine, an attempt to prepend a `\0` to open an abstract socket leads to an IOException when binding or connecting since the `GetStringUTFChars` used in the [source](https://github.com/netty/netty/blob/master/transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c#L1491) will encode the `\0` character. There could also be issues with the `strlen` and `strncpy` operations if this was moved to a different encode operation.

```
$ java -version
openjdk version "1.8.0_51"
OpenJDK Runtime Environment (build 1.8.0_51-b16)
OpenJDK 64-Bit Server VM (build 25.51-b03, mixed mode)
```

Operating System: Arch Linux 64-bit

```
$ uname -a
Linux ylt 4.1.4-1-ARCH #1 SMP PREEMPT Mon Aug 3 21:30:37 UTC 2015 x86_64 GNU/Linux
```

**Comment by normanmaurer** [comment]
@yawkat can you tell me what you mean with "abstract domain socket" ?

**Comment by yawkat** [comment]
See [this man page](http://man7.org/linux/man-pages/man7/unix.7.html). Basically, domain socket names allow `\0` bytes in them, and on linux, a domain socket starting with `\0` does not appear in the file system and is called abstract. 

[socat](http://linux.die.net/man/1/socat) supports these through a special `ABSTRACT` address family. [dbus](http://dbus.freedesktop.org/doc/dbus-specification.html)  supports them through an `abstract` property in their address system. 

Sorry for not linking the exact paragraphs, I'm on mobile.

**Comment by normanmaurer** [comment]
@yawkat so I guess using 'GetStringChars' would be right here... I'm correct ?

**Comment by yawkat** [comment]
Wouldn't that return an array of 16-bit numbers? Casting that to 8 bit would just give you lots of 0s.

I'm guessing the right way would either be decoding it as ascii or passing through the byte array directly instead of the string. Also need to replace the string operations because they rely on a trailing 0.

**Comment by normanmaurer** [comment]
@yawkat good point.

**Comment by yawkat** [comment]
Okay, had nothing else to do so I tried my hand on a fix: #4092

**Comment by normanmaurer** [comment]
Fixed by https://github.com/netty/netty/pull/4092
